obspy/io/css/tests/test_core.py

```diff
@@ -118,6 +127,32 @@ def test_nnsa_kb_core_read_via_module(self):
             tr.stats.pop('_format')
         self.assertEqual(st, self.st_result_nnsa)
 
+    def test_css_2_read_via_module(self):
+        """
+        Read files via obspy.io.css.core._read_css function.
+        Read gzipped waveforms.
+        """
+        # 1
+        st = _read_css(self.filename_css_2)
+        # _format entry is not present when using low-level function
+        for tr in self.st_result_css:
+            tr.stats.pop('_format')
+        self.assertEqual(st, self.st_result_css)
+
+    def test_css_3_read_via_module(self):
+        """
+        Read files via obspy.io.css.core._read_css function.
+        Warns that waveform files are missing.
+        """
+        # 1
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always", UserWarning)
+            _read_css(self.filename_css_3)
+            # Two waveform files are missing
+            assert len(w) == 2
+            assert "missing_dtype_c0" in str(w[0].message)
+            assert "missing11155" in str(w[1].message)
```

Comment:
There's always a (slight) chance that this test might break later on in case an unexpected warning pops up (e.g. a simple deprection warning in numpy or some other dependency). We have a helper context wrapper with sophisticated testing of warnings, e.g. used here:

https://github.com/obspy/obspy/blob/e3b19478c2a63ab714b80c0f2efcff9e8b6b9cbb/obspy/io/stationxml/tests/test_stationxml.py#L283-L289

Reply:
Thanks for pointing to the CatchAndAssertWarnings class and recommending good test code.
However, I do not have the opportunity to use it now, because I have returned to the original stricter solution, where the missing data segment is not tolerated by the warning message, but causes an error message and terminates the procedure. While this can interrupt many hours of calculation, it is the user's responsibility to keep the data in a consistent state.
It's not up to me to change the tolerance level of data errors in the core code.